lib/Dialect/Polynomial/Transforms/Passes.td

```diff
@@ -0,0 +1,20 @@
+#ifndef LIB_DIALECT_POLYNOMIAL_TRANSFORMS_PASSES_TD_
+#define LIB_DIALECT_POLYNOMIAL_TRANSFORMS_PASSES_TD_
+
+include "mlir/Pass/PassBase.td"
+
+def NTTRewrites : Pass<"ntt-rewrites"> {
+  let summary = "Rewrites polynomial operations to their NTT equivalents";
+  let description = [{
+    Applies a set of rewrites to convert polynomial operations to their
```

Comment:
Are there other rewrites of polynomial operations in terms of NTT? Maybe you're thinking of how you can do a convolution of two vectors in a coefficient encoding using NTT?

If there are not others, let's just name this pass more directly what it is, something like `convert-polynomial-mul-to-ntt`. You can still keep the patterns in `NTTRewrites.td`, and just add the `NTTRewritePolyMul` pattern by name to the pattern set instead of adding all generated patterns.

My reasoning: `ntt-rewrites` is too vague. I would guess it means "optimizing existing NTT ops." Also the handling of poly mul is somehow a very distinguished part of FHE, so I want it to be easy to find :)

Reply:
I agree with the reasoning to be more specific.

I had left it vague to allow for other NTT-related optimizations to also be applied in this pass.